### PR TITLE
Add redux and redux-saga

### PR DESCRIPTION
**Pull request by Darmody**
http://redux.js.org

### CODE REVIEW

File: src/store/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { createStore, applyMiddleware, compose } from 'redux'
+import rootReducer from '../reducers'
+import middlewares from '../middlewares'
+
+if (process.env.NODE_ENV === 'development') {
+  middlewares.push(require('redux-logger')())
+}
+
+const finalCreateStore = compose(
+  applyMiddleware(...middlewares)
```
Comment (calebx) [inline src/store/configureStore.js]: `applyMiddleware(...middlewares),` ?

File: src/store/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { createStore, applyMiddleware, compose } from 'redux'
+import rootReducer from '../reducers'
+import middlewares from '../middlewares'
+
+if (process.env.NODE_ENV === 'development') {
+  middlewares.push(require('redux-logger')())
```
Comment (calebx) [inline src/store/configureStore.js]: why not move this log into `middlewares/index.js`

File: src/sagas/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function* saga() {
+  yield []
```
Comment (calebx) [inline src/sagas/index.js]: oh, yeah

**Comment by calebx** [comment]
so fancy 👍

File: src/store/configureStore.js (hunk 1 above)
Comment (Darmody) [inline src/store/configureStore.js]: yeah

File: src/store/configureStore.js (hunk 2 above)
Comment (Darmody) [inline src/store/configureStore.js]: u are right

**Comment by Darmody** [comment]
image

Add redux-dev-tools. Install [chrome-extension](https://github.com/zalmoxisus/redux-devtools-extension ) to use it.

**Pull request merged**